### DISCUSSION TITLE
how to acknowledge on message after listing from SQS rather than deleting the message from queue in AWS JAVA SDK v2

### DISCUSSION THREAD

**Discussion by pjonnlagadda** [aws/aws-sdk-java-v2#4959] [Q&A]
My requirement is to acknowledge the message after listing from SQS rather deleting using Java SDK v2. I am referring the below page where suggesting to delete it after listen or read from queue

https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/examples-sqs-messages.html

SQSSession.UNORDERED_ACKNOWLEDGE?? how to use this in V2?
https://stackoverflow.com/questions/46863834/acknowledge-message-in-sqs-queue

Can you help on this ONLY acknowledge rather than deleing it

**Comment by debora-ito** [comment]
Answered in the GH Issue you submitted: you need to use the library `com.amazonaws:amazon-sqs-java-messaging-lib` to have access to SQSSession. See my full answer here: https://github.com/aws/aws-sdk-java-v2/issues/4981#issuecomment-1973745515
